RuntimeDetour/Platforms/Runtime/DetourRuntimeNETCorePlatform.cs

```diff
@@ -1,24 +1,158 @@
-    public
-#endif
-    class DetourRuntimeNETCorePlatform : DetourRuntimeNETPlatform {
-
-        protected override void DisableInlining(MethodBase method, RuntimeMethodHandle handle) {
-            // https://github.com/dotnet/runtime/blob/89965be3ad2be404dc82bd9e688d5dd2a04bcb5f/src/coreclr/src/vm/method.hpp#L178
-            // mdcNotInline = 0x2000
-            // References to RuntimeMethodHandle (CORINFO_METHOD_HANDLE) pointing to MethodDesc
-            // can be traced as far back as https://ntcore.com/files/netint_injection.htm
-
-            // FIXME: Take a very educated guess regarding the offset to m_wFlags in MethodDesc.
-        }
-    }
-}
+﻿using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Reflection.Emit;
+using MonoMod.Utils;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Diagnostics;
+using System.IO;
+using MonoMod.Common.RuntimeDetour.Platforms.Runtime;
+
+namespace MonoMod.RuntimeDetour.Platforms {
+#if !MONOMOD_INTERNAL
+    public
+#endif
+    class DetourRuntimeNETCorePlatform : DetourRuntimeNETPlatform {
+
+        // All of this stuff is for JIT hooking in RuntimeDetour so we can update hooks when a method is re-jitted
+        [UnmanagedFunctionPointer(CallingConvention.StdCall)]
+        private delegate IntPtr d_getJit();
+        private static d_getJit getJit;
+
+        protected static IntPtr GetJitObject() {
+            if (getJit == null) {
+                // To make sure we get the right clrjit, we enumerate the process's modules and find the one 
+                //   with the name we care aboutm, then use its full path to gat a handle and load symbols.
+                Process currentProc = Process.GetCurrentProcess();
+                ProcessModule clrjitModule = currentProc.Modules.Cast<ProcessModule>()
+                    .FirstOrDefault(m => Path.GetFileNameWithoutExtension(m.FileName).EndsWith("clrjit"));
+                if (clrjitModule == null)
+                    throw new PlatformNotSupportedException();
+
+                if (!DynDll.TryOpenLibrary(clrjitModule.FileName, out IntPtr clrjitPtr))
+                    throw new PlatformNotSupportedException();
+
+                try {
+                    getJit = clrjitPtr.GetFunction(nameof(getJit)).AsDelegate<d_getJit>();
+                } catch {
+                    DynDll.CloseLibrary(clrjitPtr);
+                    throw;
+                }
+            }
+
+            return getJit();
+        }
+
+        private static bool? isNet5Jit = null;
+
+        protected static Guid GetJitGuid(IntPtr jit) {
+            Guid guid;
+            if (isNet5Jit == null) {
+                // if we don't know, we first try index 2, because it is harmless on .NET Core 3, even passing a pointer
+                // on second thought, it probably isn't on x86 because of callee stack cleanup, but idk how to make this work otherwise

```

Comment:
We should be able to determine this using the `ProcessModule` itself.

```cs
clrjitModule.FileVersionInfo.ProductMajorPart > 5
```

Reply:
It didn't even occur to me to check the file metadata for this.